### ISSUE TITLE
Bug: Map: Expected mapDiv of type Element but was passed undefined.

### ISSUE THREAD

**Issue by shayneo** [diegoazh/gmap-vue#122]
**Describe the bug**
"Map: Expected mapDiv of type Element but was passed undefined." error is thrown when navigating from one page with a map, to another page with a different map.

**To Reproduce**
1. add a gmap on a page, give it a ref and id like `mymap`
2. add another gmap to a different page, give it a different ref and id like`myothermap`

**Expected behavior**
`myothermap` loads without throwing error

**Current behavior**
"Map: Expected mapDiv of type Element but was passed undefined."  is thrown when navigating to page with `myothermap`

**Screenshots**
n/a

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] MacOS
 - Browser [e.g. chrome, safari] firefox
 - Version [e.g. 22] latest

**Smartphone (please complete the following information):**
n/a

**Additional context**
might have something to do with calling $mapobject.destroy()

## Versions

- Node: 14
- NPM: 6.14.8
- Yarn: n.a

## Package manager

- [x] NPM
- [ ] Yarn

## Plugin version

- [x] 1.4.x
- [ ] 1.2.x
- [ ] 1.0.x

**Comment by shayneo** [comment]
I have no idea what is wrong with my issue.

**Comment by shayneo** [comment]
Thanks for reopening, any ideas on the actual issue?

**Comment by clp-atoms** [comment]
I have the same problem, there is news on how to solve it?

**Comment by kruebling** [comment]
@clp-atoms @shayneo 
Ran into this issue as well, turned out to be a problem with how my components were structured.

```
// ParentComponent.vue

<template>
  <div>
    <SomeLoadingComponent v-if="loading" />
    <div v-else>
      <GMapVueComponent />
      <SomeOtherComponent />
    </div>
  </div>
</template>
<script>
...
```

It seemed that in my GMapVueComponent, the `$mapPromise` was resolving before the loading state had finished in the ParentComponent to trigger the v-else block. Everything worked as expected, but I got a couple errors every time I loaded the page. Looked like they were coming from these two places. One for an undefined `$mapObject` and the other for an undefined element.

- https://github.com/diegoazh/gmap-vue/blob/master/packages/gmap-vue/src/components-implementation/map.js#L170
- https://github.com/diegoazh/gmap-vue/blob/master/packages/gmap-vue/src/components-implementation/map.js#L143

Passing down the conditional logic as props to the GMapVueComponent and not wrapping it's usage in a v-else block seemed to clear everything up. Hope that's helpful to anyone else running into this!
